dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1307,7 +1309,16 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Auth User user,
                     existingTag.setDirtyBit(true);
                 }
 
+                boolean wasFrozen = existingTag.isFrozen();
                 existingTag.updateByUser(version);
+                boolean nowFrozen = existingTag.isFrozen();
+                // If version is snapshotted on this update, grab and store image information
+                if (!wasFrozen && nowFrozen) {
+                    LanguageHandlerInterface lInterface = LanguageHandlerFactory.getInterface(w.getFileType());
+                    String toolsJSONTable = lInterface.getContent(w.getWorkflowPath(), getMainDescriptorFile(existingTag).getContent(), extractDescriptorAndSecondaryFiles(existingTag), LanguageHandlerInterface.Type.TOOLS, toolDAO);
+                    Set<Image> images = lInterface.getImagesFromRegistry(toolsJSONTable);
+                    existingTag.getImages().addAll(images);
```

Comment:
I thought we discussed not allowing snapshotting if the images were parameterizable. Is that covered in another ticket?

Also, if there are errors in getting the images, this will fail silently and there will be no indication to the user. If there is an error, should we even allow snapshotting? Not sure...

Reply:
It can be in another ticket. My understanding is CWL only accepts strings in a `dockerPull` so WDL is the one with parameterizeable images. 

My thoughts on if there was an error getting an image was that it was okay to let the process continue. There may be cases where we miss docker refs entirely or are unable to properly parse it (like how wdl docker refs allow variables, doesn't seem easy for use to handle?). 
But maybe we should indicate to the user we weren't able to process certain images.